### ISSUE TITLE
module is not compatible with setuptools v34

### ISSUE THREAD

**Issue by vchepkov** [stankevich/puppet-python#361]
Unfortunately, due to changes in setuptools http://setuptools.readthedocs.io/en/latest/history.html#v34-0-0
virtualenv resource doesn't work anymore, due to missing dependencies:

```
# virtualenv /opt/virtualenv
New python executable in /opt/virtualenv/bin/python
Installing Setuptools..............................................................................................................................................................................................................................done.
Installing Pip.....................................................................................................................................................................................................................................................................................................................................done.

# /opt/virtualenv/bin/pip install --upgrade setuptools
Downloading/unpacking setuptools from https://pypi.python.org/packages/64/88/d434873ba1ce02c0ed669f574afeabaeaaeec207929a41b5c1ed947270fc/setuptools-34.1.0.zip#md5=a2f819c503b49781b1a15cfb4d49c974
  Downloading setuptools-34.1.0.zip (621kB): 621kB downloaded
  Running setup.py egg_info for package setuptools
    Traceback (most recent call last):
      File "<string>", line 3, in <module>
      File "setuptools/__init__.py", line 10, in <module>
        from six.moves import filter, map
    ImportError: No module named six.moves
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 3, in <module>

  File "setuptools/__init__.py", line 10, in <module>

    from six.moves import filter, map

ImportError: No module named six.moves

----------------------------------------
Cleaning up...
Command python setup.py egg_info failed with error code 1 in /opt/virtualenv/build/setuptools
Storing complete log in /root/.pip/pip.log
```

**Comment by vchepkov** [comment]
It looks like if you upgrade pip first, then you can upgrade setuptools

**Comment by vchepkov** [comment]
darn, didn't notice #359, well, same idea

**Comment by ThiefMaster** [comment]
A release containing this fix would be great.

**Comment by fenixwanamaker** [comment]
Indeed, a new release including the latest setuptools compatibility would be nice ( this issue currently breaks [puppetboard](https://github.com/voxpupuli/puppetboard) automatic installation, among others )

**Comment by elconas** [comment]
:+1:  We also needs this. Otherwise virtualenv can not be used.

**Comment by vchepkov** [comment]
fixed by #388
